client/asset/eth/multirpc.go

```diff
@@ -803,7 +806,19 @@ func (m *multiRPCClient) withOne(providers []*provider, f func(*provider) error,
 		readyProviders = providers
 	}
 	for _, p := range readyProviders {
-		err := f(p)
+		ctx, cancel := context.WithTimeout(ctx, defaultRequestTimeout)
+		// Fetching the best header will check that either the
+		// provider's cached header is not too old or that a
+		// newly fetched header is not too old. If it is too
+		// old that indicates the provider is not in sync and
+		// should not be used.
+		if _, err := p.bestHeader(ctx, m.log, m.net); err != nil {
+			cancel()
+			m.log.Warnf("Problem getting best header: %s.", err)
+			continue
+		}
+		err := f(ctx, p)
+		cancel()
 		if err == nil {
 			break
 		}
```

Comment:
Since a `readyProvider` could potentially never run now, we need to do two things. 1) change this to 

```Go
if err == nil {
    return nil
}
```

and then 2) if we make it out of the `readyProviders` loop, it now means that all providers failed or weren't up to date, so we would return an error.

I actually think that `break`ing when `err == nil` was incorrect anyway, since `superError` could have been set in a previous iteration.

Reply:
Done.